modules/pricer/src/main/java/com/opengamma/strata/pricer/impl/volatility/smile/SabrHaganNormalVolatilityFormula.java

```diff
@@ -0,0 +1,416 @@
+/*
+ * Copyright (C) 2022 - present by OpenGamma Inc. and the OpenGamma group of companies
+ *
+ * Please see distribution for license.
+ */
+package com.opengamma.strata.pricer.impl.volatility.smile;
+
+import static com.opengamma.strata.math.MathUtils.nearZero;
+
+import java.io.Serializable;
+import java.lang.invoke.MethodHandles;
+
+import org.joda.beans.ImmutableBean;
+import org.joda.beans.MetaBean;
+import org.joda.beans.TypedMetaBean;
+import org.joda.beans.gen.BeanDefinition;
+import org.joda.beans.impl.light.LightMetaBean;
+
+import com.opengamma.strata.basics.value.ValueDerivatives;
+import com.opengamma.strata.collect.ArgChecker;
+import com.opengamma.strata.collect.array.DoubleArray;
+import com.opengamma.strata.market.ValueType;
+import com.opengamma.strata.pricer.model.SabrVolatilityFormula;
+
+/**
+ * Formulas related to the SABR implied normal volatility function.
+ * <p>
+ * Only the "beta = 0" versions take strike/forward with negative values. The other formulas have a barrier at 0.
+ * <p>
+ * Reference: Hagan, P.; Kumar, D.; Lesniewski, A. & Woodward, D. "Managing smile risk", Wilmott Magazine, 2002, September, 84-108
+ * Note: Formula references are related to the preprint version, e.g. available at https://www.researchgate.net/publication/235622441_Managing_Smile_Risk
+ */
+@BeanDefinition(style = "light")
+public final class SabrHaganNormalVolatilityFormula
+    implements SabrVolatilityFormula, ImmutableBean, Serializable {
```

Comment:
Does this need to be a bean?

Reply:
No it does not; there is no data in it. I copied the other version (`SabrHaganVolatilityFunctionProvider`) which is a Bean. Probably we should also remove the Bean feature on the other one.